mesh-multi-zone/kong-mesh-multi-zone.md

```diff
@@ -740,6 +750,217 @@ destinations:
+```
+
+## Health checks
+
+This step doesn't affect the deployment, it is just to avoid the Health check errors.
+
+It seems that the current version of Lua that uses the proxy containers doesn't support HTTP/2.0 and this error keeps
+showing up in the logs
+
+```text
+127.0.0.1 - - [29/Apr/2022:08:42:59 +0000] "GET /status HTTP/2.0" 500 42 "-" "Go-http-client/2.0"
+2022/04/29 08:43:02 [error] 2073#0: *135 [lua] api_helpers.lua:526: handle_error(): /usr/local/share/lua/5.1/lapis/application.lua:424: /usr/local/share/lua/5.1/kong/api/routes/health.lua:52: http2 requests not supported yet
+stack traceback:
+       [C]: in function 'capture'
+       /usr/local/share/lua/5.1/kong/api/routes/health.lua:52: in function 'fn'
+       /usr/local/share/lua/5.1/kong/api/api_helpers.lua:293: in function 'fn'
+        /usr/local/share/lua/5.1/kong/api/api_helpers.lua:293: in function </usr/local/share/lua/5.1/kong/api/api_helpers.lua:276>
+```
+
+To avoid this error we can patch the deployment to not use HTTP/2.0
+
+```bash
+oc patch deployment -n kong kong-kong -p "{\"spec\": { \"template\" : { \"spec\" : {\"containers\":[{\"name\":\"proxy\",\"env\": [{ \"name\" : \"KONG_PROXY_LISTEN\", \"value\":                                   
+\"0.0.0.0:8000, 0.0.0.0:8443 ssl\" }, { \"name\" : \"KONG_ADMIN_LISTEN\", \"value\": \"127.0.0.1:8444 ssl\" }]}]}}}}"
+```
+
+## Proxy traffic to our demo app
+
+Deploy the [kuma-demo application](../kong-mesh.md#kuma-demo-application) on the `kuma-demo` namespace.
+
+Create the kuma-demo ingress. We're creating a specific route in the kong namespace for setting
+a dedicated hostname for this ingress.
+
+```bash
+OCP_DOMAIN=`oc get ingresses.config/cluster -o jsonpath={.spec.domain}`
+wget https://raw.githubusercontent.com/RHEcosystemAppEng/kong-discovery/main/kic/kuma-demo-ingress.yaml
+sed -e "s/\${i}/1/" -e "s/\$OCP_DOMAIN/$OCP_DOMAIN/" kuma-demo-ingress.yaml | oc apply -f -
+```
+
+You can now access your application:
+
+```bash
+http `oc get route demo-app -n kong --template='{{ .spec.host }}'`/
+```
+
+### Using Plugins
+
+There are 2 types of plugins:
+
+- KongPlugins: Can be used by resources in the same namespace
+- KongClusterPlugins: Can be used cluster wide
+
+### RateLimiting example
+
+Create a KongPlugin and update the Ingress to use it.
+
+```bash
+oc apply -f https://raw.githubusercontent.com/RHEcosystemAppEng/kong-discovery/main/kic/simple-rate-limiting.yaml
+oc annotate ingress demo-app-ingress --overwrite -n kuma-demo konghq.com/plugins=rate-free-tier
+```
+
+After that, you can see in the HTTP headers the rate limiting information:
+
+```bash
+$ http `oc get route demo-app -n kong --template='{{ .spec.host }}'`
+HTTP/1.1 200 OK
+...
+ratelimit-limit: 10
+ratelimit-remaining: 9
+ratelimit-reset: 54
+x-ratelimit-limit-minute: 10
+x-ratelimit-remaining-minute: 9
+```
+
+### Advanced RateLimiting example
```

Comment:
This is not relevant to the scope of this document, you should document how to reach the service through the ingress controller which is not here.

Reply:
Removed unused section